api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+package org.pmiops.workbench.access;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
+import java.sql.Timestamp;
+import java.time.Clock;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.pmiops.workbench.actionaudit.auditors.UserServiceAuditor;
+import org.pmiops.workbench.actionaudit.targetproperties.BypassTimeTargetProperty;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.UserAccessModuleDao;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.model.DbAccessModule;
+import org.pmiops.workbench.db.model.DbAccessModule.AccessModuleName;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbUserAccessModule;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.exceptions.ForbiddenException;
+import org.pmiops.workbench.model.AccessBypassRequest;
+import org.pmiops.workbench.model.AccessModule;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AccessModuleServiceImpl implements AccessModuleService {
+  private static final Logger logger = Logger.getLogger(AccessModuleServiceImpl.class.getName());
+
+  private final Provider<List<DbAccessModule>> dbAccessModulesProvider;
+  private final Clock clock;
+
+  private final UserAccessModuleDao userAccessModuleDao;
+  private final UserDao userDao;
+  private final UserServiceAuditor userServiceAuditor;
+  private final Provider<WorkbenchConfig> configProvider;
+
+  private static final BiMap<AccessModule, AccessModuleName> CLIENT_TO_STORAGE_ACCESS_MODULE =
+      ImmutableBiMap.<AccessModule, AccessModuleName>builder()
+          .put(AccessModule.TWO_FACTOR_AUTH, AccessModuleName.TWO_FACTOR_AUTH)
+          .put(AccessModule.ERA_COMMONS, AccessModuleName.ERA_COMMONS)
+          .put(AccessModule.COMPLIANCE_TRAINING, AccessModuleName.RT_COMPLIANCE_TRAINING)
+          .put(AccessModule.RAS_LINK_LOGIN_GOV, AccessModuleName.RAS_LOGIN_GOV)
+          .put(AccessModule.DATA_USE_AGREEMENT, AccessModuleName.DATA_USER_CODE_OF_CONDUCT)
+          .put(AccessModule.PUBLICATION_CONFIRMATION, AccessModuleName.PUBLICATION_CONFIRMATION)
+          .put(AccessModule.PROFILE_CONFIRMATION, AccessModuleName.PROFILE_CONFIRMATION)
+          .build();
+
+  private static final BiMap<BypassTimeTargetProperty, AccessModuleName>
```

Comment:
I was thinking if this should go into DbStorageEnum or here, then I feel this is a better place as it is not really a short, and the service can be the central place for AccessModule related utils...